### ISSUE TITLE
[GRPO Trainer] Accuracy reward stays 0

### ISSUE THREAD

**Issue by Revist** [huggingface/trl#3903]
### Reproduction

Hi all,

Running the command below with GRPO trainer 

```python
accelerate launch \
    --config_file examples/accelerate_configs/deepspeed_zero3.yaml \
    examples/scripts/grpo_vlm.py \
    --model_name_or_path Qwen/Qwen2.5-VL-3B-Instruct \
    --output_dir grpo-Qwen2.5-VL-3B-Instruct \
    --learning_rate 1e-5 \
    --gradient_checkpointing \
    --torch_dtype bfloat16 \
    --max_prompt_length 2048 \
    --max_completion_length 1024 \
    --use_vllm \
    --vllm_mode colocate \
    --use_peft \
    --lora_target_modules "q_proj", "v_proj" \
    --log_completions
```

I have observed the following behaviour:

The reward for format converges to 1 at the end of training while reward for accuracy stays 0 the whole time. 

Is this a bug?

### System Info

- Platform: Linux-5.14.0-427.50.1.el9_4.x86_64-x86_64-with-glibc2.34
- Python version: 3.12.11
- TRL version: 0.22.0.dev0+f23543f
- PyTorch version: 2.7.0
- accelerator(s): NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB, NVIDIA A100-SXM4-80GB
- Transformers version: 4.53.3
- Accelerate version: 1.9.0
- Accelerate config: not found
- Datasets version: 4.0.0
- HF Hub version: 0.33.4
- bitsandbytes version: not installed
- DeepSpeed version: 0.17.2
- Diffusers version: not installed
- Liger-Kernel version: not installed
- LLM-Blender version: not installed
- OpenAI version: 1.90.0
- PEFT version: 0.16.0
- vLLM version: 0.9.2

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete

**Comment by qgallouedec** [comment]
Hey @Revist, sorry for the very late reply, we're catching up on the backlog.

This isn't a trainer bug. `accuracy_reward` grades answers with `math_verify`, which only credits LaTeX/`\boxed{}` math: it parses with `try_extract_without_anchor=False`. But the example's dataset `lmms-lab/multimodal-open-r1-8k-verified` stores its ground truth as a multiple-choice letter inside `<answer>A</answer>` tags, so `math_verify` can't parse the gold (it returns `[]`, or scrapes a spurious number out of the reasoning text) and the accuracy reward is ~0 by construction. 

`think_format_reward` only checks the `<think></think>` structure, which is easy, so it converges to 1, exactly what you saw.

Since you're running with `--log_completions`, you can confirm this directly: check the logged `solution` / `gold_parsed` / `answer_parsed` columns and you'll see `gold_parsed` is empty or garbage. To actually train accuracy on a multiple-choice VQA set you need a reward that extracts the `<answer>...</answer>` letter and compares it; or switch to a math dataset where `accuracy_reward` applies and prompt the model to put its final answer in `\boxed{}`.

(Minor unrelated note: `--lora_target_modules "q_proj", "v_proj"` is tokenized by the shell into `q_proj,` and `v_proj`, so only `v_proj` actually gets LoRA; drop the comma/quotes.)
